pyproject.toml

```diff
@@ -0,0 +1,80 @@
+[build-system]
+requires = ['hatchling', 'hatch-vcs']
+build-backend = 'hatchling.build'
+
+[project]
+name = 'fmf'
+authors = [
+    { name = 'Petr Splichal', email = 'psplicha@redhat.com' },
+]
+maintainers = [
+    { name = 'Petr Splichal', email = 'psplicha@redhat.com' },
+]
+description = 'Flexible Metadata Format'
+readme = 'README.rst'
+license = 'GPL-2.0-or-later'
+license-files = { paths = ['LICENSE'] }
+requires-python = '>=3.9'
+classifiers = [
+    'Natural Language :: English',
+    'Programming Language :: Python :: 3.9',
+    'Programming Language :: Python :: 3.10',
+    'Programming Language :: Python :: 3.11',
+    'Programming Language :: Python :: 3.12',
+    'Topic :: Utilities',
+]
+keywords = [
+    'metadata',
+    'testing',
+]
+dependencies = [
+    'ruamel.yaml',
+    'filelock',
+    'jsonschema',
+]
+dynamic = ['version']
+
+[project.urls]
+Homepage = 'https://github.com/psss/fmf'
+
+[project.optional-dependencies]
```

Comment:
I'm not sure if you want to introduce these subpackages, as I'm guessing these dependencies are aimed at developers, not end users. Might be worth adding them as hatch environments, like in tmt.  

Additionally, I'd just do one "dev" and one "docs", since there aren't many deps.

Reply:
The reason for this split is that each environment is used in different parts of the CI
- `tests` in the `spec` file
- `tests-cov` in the github CI
- `docs` in RTD
- `dev` for the developers to make sure `pre-commit` and other tools are used (maybe should add `mypy` there)

It might be that the runtime could be split in different user-level dependencies, but I am not sure which would be currently.

But `all` is excessive and ambiguous, I should remove it. You can comma separate the optional dependencies anyway.

Reply:
>     * `tests` in the `spec` file

Is the check in specfile necessary, given the fact the tests are being run in the CI workflow anyway?
 
>     * `tests-cov` in the github CI
> 
>     * `docs` in RTD
> 
>     * `dev` for the developers to make sure `pre-commit` and other tools are used (maybe should add `mypy` there)
> 

Right, nothing against splitting them if it saves dependency installation here and there, my comment was mostly aimed at "optional extras" vs "hatch environments". Let's say, for tests-cov there can be:
```
[tool.hatch.envs.cov]
dependencies [
    'fmf[tests]',
    'pytest-cov',
]
[tool.hatch.envs.cov.scripts]
xml = ["pytest --cov --cov-report=xml"]
```
and in CI, you'd just run `harch run cov:xml`

Dev could be similar to tmt: https://tmt.readthedocs.io/en/latest/contribute.html#develop

Reply:
> Is the check in specfile necessary, given the fact the tests are being run in the CI workflow anyway?

Yes, it's a guideline recommendation. It can picks up on issues with the OS packages, while the CI generally always takes the most recent (by design).

> Right, nothing against splitting them if it saves dependency installation here and there, my comment was mostly aimed at "optional extras" vs "hatch environments". Let's say, for tests-cov there can be:

Fair point. I could see usages for this, but I think in this project it is not needed until there are user-level optional dependencies, which I don't think will happen soon.

Also note that there are tools that scrape for PEP621 to get a dependency graph and so on, which would not work with specific backend formats.